bindings/src/main/java/io/pravega/storage/hdfs/HDFSStorage.java

```diff
@@ -654,5 +667,77 @@ private int readInternal(SegmentHandle handle, byte[] buffer, long offset, int b
         return length;
     }
 
+    @Override
+    public Iterator<SegmentProperties> listSegments() throws IOException {
+        try {
+            return new HDFSSegmentIterator(this.fileSystem.listStatus(new Path(config.getHdfsRoot() + Path.SEPARATOR)),
+                    fileStatus -> {
+                        String fileName = fileStatus.getPath().getName();
+                        int index = fileName.lastIndexOf(PART_SEPARATOR);
+                        if (fileName.endsWith(PART_SEPARATOR + SEALED)) {
+                            return true;
+                        }
+                        try {
+                            Long.parseLong(fileName.substring(index + 1));
+                        } catch (NumberFormatException nfe) {
+                            return false;
+                        }
+                        return true;
+                    });
+        } catch (Exception e) {
+            log.error("Exception occurred while listing the segments.", e);
+            throw e;
+        }
+    }
+
+    /**
+     * Iterator for segments in HDFS Storage.
+     */
+    private static class HDFSSegmentIterator implements Iterator<SegmentProperties> {
+        private final Iterator<SegmentProperties> results;
+
+        HDFSSegmentIterator(FileStatus[] results, java.util.function.Predicate<FileStatus> patternMatchPredicate) {
+                this.results = Arrays.asList(results).stream()
+                        .filter(patternMatchPredicate)
+                        .map(this::toSegmentProperties)
+                        .iterator();
+        }
+
+        public SegmentProperties toSegmentProperties(FileStatus fileStatus) {
+            try {
+                boolean isSealed = isSealed(fileStatus.getPath());
+                return StreamSegmentInformation.builder()
+                        .name(getSegmentNameFromPath(fileStatus.getPath()))
+                        .length(fileStatus.getLen())
+                        .sealed(isSealed).build();
+            } catch (Exception e) {
+                log.error("Exception occurred while transforming the object into SegmentProperties.");
+                return null;
+            }
+        }
+
+        /**
+         * Method to check the presence of next element in the iterator.
+         * @return true if the next element is there, else false.
+         */
+        @Override
+        public boolean hasNext() {
+            return results.hasNext();
+        }
+
+        /**
+         * Method to return the next element in the iterator.
+         * @return A newly created StreamSegmentInformation class.
+         * @throws NoSuchElementException in case of an unexpected failure.
+         */
+        @Override
+        public SegmentProperties next() throws NoSuchElementException {
+            try {
+                return results.next();
+            } catch (NoSuchElementException e) {
```

Comment:
There is no need for this try-catch as the exception will automatically bubble up.